File: v2/pkg/js/libs/fs/fs.go
package fs

import (
	"os"

	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/protocolstate"
)

// ListDir lists directory contents within permitted paths
// itemType can be used to filter type of results
// allowed values are: file, dir, all
func ListDir(path string, itemType string) ([]string, error) {
	finalPath, err := protocolstate.NormalizePath(path)
	if err != nil {
		return nil, err
	}
	values, err := os.ReadDir(finalPath)
	if err != nil {
		return nil, err
	}
	var results []string
	for _, value := range values {
		if itemType == "file" && value.IsDir() {
			continue
		}
		if itemType == "dir" && !value.IsDir() {
			continue
		}
		results = append(results, value.Name())
	}
	return results, nil
}

// ReadFile reads file contents within permitted paths
func ReadFile(path string) ([]byte, error) {
	finalPath, err := protocolstate.NormalizePath(path)
	if err != nil {
		return nil, err
	}
	bin, err := os.ReadFile(finalPath)
	return bin, err
}

// ReadFileAsString reads file contents within permitted paths
// and returns content as string
func ReadFileAsString(path string) (string, error) {
	bin, err := ReadFile(path)
	if err != nil {
		return "", err
	}
	return string(bin), nil
}

File: pkg/js/devtools/tsdocs/cmd/tsdocs/src/fs.ts

/**
 * ListDir lists all files and directories within a path
 * depending on the itemType provided
 * itemType can be any one of ['file','dir','all']
* @throws {Error} - if the operation fails
 */
export function ListDir(path: string, itemType: string): string[] | null {
    return null;
}


/**
 * ReadFile reads file contents within permitted paths
* @throws {Error} - if the operation fails
 */
export function ReadFile(path: string): Uint8Array | null {
    return null;
}


/**
 * ReadFileAsString reads file contents within permitted paths
 * and returns content as string
* @throws {Error} - if the operation fails
 */
export function ReadFileAsString(path: string): string | null {
    return null;
}


/**
 * ReadFilesFromDir reads all files from a directory
 * and returns a array with file contents of all files
* @throws {Error} - if the operation fails
 */
export function ReadFilesFromDir(dir: string): string[] | null {
    return null;
}

